src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +116,341 @@ def execute(self, reqStatus):
+            self.updateReportDict(summary, "archived_num_requests", archived_num_requests)
+        except Exception as ex:
+            msg = "Unknown exception while running MSRuleCleaner thread Error: %s"
+            self.logger.exception(msg, str(ex))
+            self.updateReportDict(summary, "error", msg)
+
         return summary
 
+    def _execute(self, reqRecords):
+        """
+        Executes the MSRuleCleaner pipelines based on the workflow status
+        :param reqList: A list of RequestRecords to work on
+        :return:        a tuple with:
+                            number of properly cleaned requests
+                            number of processed workflows
+                            number of archived workflows
+        """
+        # NOTE: The Input Cleanup, the Block Level Cleanup and the Archival
+        #       Pipelines are executed sequentially in the above order.
+        #       This way we assure ourselves that we archive only workflows
+        #       that have accomplished the needed cleanup
+
+        clean_num_requests = 0
+        total_num_requests = 0
+
+        # Call the workflow dispatcher:
+        for reqName, req in reqRecords.items():
+            wflow = MSRuleCleanerWflow(reqRecords[reqName])
+            self._dispatchWflow(wflow)
+            msg = "\n----------------------------------------------------------"
+            msg += "\nMSRuleCleanerWflow: %s"
+            msg += "\n----------------------------------------------------------"
+            self.logger.debug(msg, pformat(wflow))
+            total_num_requests += 1
+            if self._checkClean(wflow):
+                clean_num_requests += 1
+
+        # Report the counters:
+        for pline in self.cleanuplines:
+            msg = "Workflows cleaned by pipeline: %s: %d"
+            self.logger.info(msg, pline.name, self.wfCounters['cleaned'][pline.name])
+        archived_num_requests = self.wfCounters['archived']
+        msg = "Workflows archived: %d"
+        self.logger.info(msg, self.wfCounters['archived'])
+        return total_num_requests, clean_num_requests, archived_num_requests
+
+    def _dispatchWflow(self, wflow, *args, **kwargs):
+        """
+        A function intended to dispatch a workflow (e.g based on its status)
+        through one or more functional pipelines in case there is some more
+        complicated logic involved in the order we execute them but not just
+        a sequentially
+        """
+        msg = "Called with wflow: %s"
+        self.logger.debug(msg, pformat(wflow))
+        # NOTE: The following dispatch logic is a subject to be changed at any time
+
+        # Do not clean any Resubmission, but still let them be archived
+        if wflow['RequestType'] == 'Resubmission':
+            wflow['ForceArchive'] = True
+            msg = "Skipping cleanup step for workflow: %s - RequestType is %s."
+            msg += "\nWill try to archive it directly."
```

Comment:
How about we keep all this log in the same line? It's usually helpful when scanning the service logs and debugging it.

Reply:
done